src/components/App/App.js

```diff
@@ -42,6 +51,43 @@ class App extends Component {
     }
   }
 
+  componentDidMount() {
+    if (typeof window.ethereum !== 'undefined'){
+      window.ethereum.on('chainChanged', (chainId) => {
+        if (chainId !== MAINNET_CHAIN_ID_HEX) {
+          this.setState({ showNetworkChangeModal: true, chainId: parseChainIdHex(chainId)});
```

Comment:
Just a head's up, these `setState` calls here could be called when the component is not mounted anymore resulting in a runtime error. A good practice would be removing the listeners when the component unmounts. Since this is the `App` and this app has a single route it will not get unmounted, but it might in the future if the dapp gets more complex.